src/AzureStorageQueueWorkerServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+
+namespace Likvido.Worker.AzureStorageQueue
+{
+    public static class AzureStorageQueueWorkerServiceCollectionExtensions
+    {
+        private static bool _timeoutConfigured = false;
+        private static bool _telemetryConfigured = false;
+
+        public static IServiceCollection AddMessageProcessor<TMessage, TMessageProcessor>(
+            [NotNull] this IServiceCollection serviceCollection,
+            [NotNull] Action<IServiceProvider, AzureStorageQueueWorkerOptionsBuilder> optionsAction,
+            ServiceLifetime serviceLifetime = ServiceLifetime.Scoped,
+            bool configureTelemetry = true)
+            where TMessageProcessor : IMessageProcessor<TMessage>
+        {
+            serviceCollection.TryAdd(
+                new ServiceDescriptor(
+                    typeof(AzureStorageQueueWorkerOptions<TMessageProcessor>),
+                    p => CreateAzureStorageQueueWorkerOptions<TMessageProcessor>(p, optionsAction),
+                    ServiceLifetime.Singleton));
+
+            serviceCollection.TryAdd(
+                new ServiceDescriptor(
+                    typeof(TMessageProcessor),
+                    typeof(TMessageProcessor),
+                    serviceLifetime));
+
+            serviceCollection.AddHostedService<AzureStorageQueueWorker<TMessage, TMessageProcessor>>();
+
+            if (!_timeoutConfigured)
+            {
+                _timeoutConfigured = true;
+                serviceCollection.PostConfigure<HostOptions>(o => o.ShutdownTimeout = TimeSpan.FromSeconds(20));
+            }
+            if (!_telemetryConfigured && configureTelemetry)
+            {
+                _telemetryConfigured = true;
+                serviceCollection.AddApplicationInsightsTelemetryWorkerService();
+            }
+
+            return serviceCollection;
+        }
+
+        private static AzureStorageQueueWorkerOptions<TMessageProcessor> CreateAzureStorageQueueWorkerOptions<TMessageProcessor>(
+            [NotNull] IServiceProvider applicationServiceProvider,
+            [NotNull] Action<IServiceProvider, AzureStorageQueueWorkerOptionsBuilder> optionsAction)
+        {
+            var options = new AzureStorageQueueWorkerOptions<TMessageProcessor>();
+            var builder = new AzureStorageQueueWorkerOptionsBuilder(options);
+
+            optionsAction?.Invoke(applicationServiceProvider, builder);
+
+            return options;
```

Comment:
We should verify that both `QueueName` & `AzureStorageConnectionString` has been provided, and throw an exception if not

Reply:
Alternatively, move them to the `AddMessageProcessor` extension method, to force setting them

Reply:
yep, good point
I'm still not sure. The way it works right now. This two lines will fail if anything is wrong with `AzureStorageConnectionString` or `QueueName`
https://github.com/Likvido/Likvido.Worker.AzureStorageQueue/blob/4f8a4b836b2d4aa72f13934ed3d339a4c5d60fb2/src/AzureStorageQueueWorker.cs#L66-L67

and in the following code, a consumer can control if should swallow the error or stop with a particular exit code using `_workerOptions.SetupIssueStopHostCod`
https://github.com/Likvido/Likvido.Worker.AzureStorageQueue/blob/4f8a4b836b2d4aa72f13934ed3d339a4c5d60fb2/src/AzureStorageQueueWorker.cs#L135-L138

But if you want I can add throwing an exception during init config phase but this will prevent the whole app from starting

Reply:
I would prefer throwing the exception during initialization